tests/integration/test_dbapi_integration.py

```diff
@@ -859,3 +859,13 @@ def test_info_uri(trino_connection):
     cur.fetchall()
     assert cur.info_uri is not None
     assert cur._query.query_id in cur.info_uri
+
+
+def test_use_schema(trino_connection):
+    cur = trino_connection.cursor()
```

Comment:
Please add assertions to verify the original catalog and schema before executing `USE` query.

Reply:
Should we also assert that new cursors opened after a `USE` still use the default catalog and schema specified on the connection?

Reply:
Currently it seems indeed that new cursors don't take over the session properties.

However I don't think this will match the expectation of the users:

> JDBC defines a system session and transaction model for database access. A session is the duration of one connection to the database and is handled by a JDBC Connection object.

See [oracle docs](https://docs.oracle.com/javadb/10.8.3.0/devguide/cdevconcepts38375.html#:~:text=JDBC%20defines%20a%20system%20session,by%20a%20JDBC%20Connection%20object.&text=A%20Connection%20object%20represents%20a%20connection%20with%20a%20database.)

WDYT?

Reply:
DBAPI != JDBC. We should do what DB-API says and if it doesn't say explicitly then follow what some other drivers do (psycopg for example doesn't maintain search path changes across cursors last time I checked).

Reply:
I checked the DBAPI specs, nothing mentioned about this. IMHO it's more of a general expectation. See e.g. https://stackoverflow.com/questions/8797724/mysql-concepts-session-vs-connection

Snowflake for example does apply it to the connection:

https://github.com/snowflakedb/snowflake-connector-python/blob/107c3033cb2c7c6961d06aa613e7c3a8f20ae435/src/snowflake/connector/cursor.py#L753-L759

In a lot of cases this is also not really a client thing, but something stored within the database side connection/session.

Reply:
@ebyhr , @hashhar : WDYT?

Reply:
@hovaesco : What's your view on this? I think in the PR's code we currently do some locking in `ClientSession`. IMHO this should be per connection instead of per cursor.

Reply:
It would be more useful when doing locking on connection/sesion level not per cursor which is limited and not user-friendly.